File: dotnet/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example2/Light.cs
namespace CommandDesignPattern.Example2;

public class Light
{
    private string location;
    private bool isOn;

    public Light(string location)
    {
        this.location = location;
        this.isOn = false;
    }

    public void TurnOn()
    {
        isOn = true;
        Console.WriteLine($"{location} ışığı açıldı.");
    }

    public void TurnOff()
    {
        isOn = false;
        Console.WriteLine($"{location} ışığı kapandı.");
    }

    public void IncreaseBrightness()
    {
        Console.WriteLine($"{location} ışığının parlaklığı artırıldı.");
    }

    public void DecreaseBrightness()
    {
        Console.WriteLine($"{location} ışığının parlaklığı azaltıldı.");
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example2/src/Light.ts
/**
 * Receiver — The light device.
 */
export class Light {
  constructor(private readonly _location: string) {}

  turnOn(): void {
    console.log(`${this._location} light turned on.`);
  }

  turnOff(): void {
    console.log(`${this._location} light turned off.`);
  }

  increaseBrightness(): void {
    console.log(`${this._location} light brightness increased.`);
  }

  decreaseBrightness(): void {
    console.log(`${this._location} light brightness decreased.`);
  }
}
